pypa/setuptools
# Merge with pypa/distutils@b65aa40

- Convert test_cmd to pytest
- Convert BasePyPIRCCommandTestCase to pytest
- Convert RegisterTestCase to pytest
- ⚫ Fade to black.
- Use jaraco.path to generate a tree.
- Port sdist tests to pytest
- Convert TestUpload to pytest
- Convert TestUpload to parametrized test. Remove dependence on unittest for PyPIRC tests.
- Implement HTTP 400 error as a pytest.param to avoid collection error. Ref pytest-dev/pytest#10184.
- Convert core tests to pytest
- ⚫ Fade to black.
- Convert TestVersion to pytest.
- Convert TestConfig to pytest
- Convert TestCygwinCCompiler to pytest
- Convert TestDirUtil to pytest
- Convert TestFileUtil to pytest
- Convert TestSysconfig to pytest
- ⚫ Fade to black.
- 👹 Feed the hobgoblins (delint).
- Remove patching of uname.
- Convert TestUtil to pytest
- Convert TestUnixCCompiler to pytest
- Prefer pytest for skip
- Convert more tests to pytest
- Prefer pytest for skip
- Prefer pytest for skip
- Convert TestBuild to pytest
- Prefer pytest for skip
- Prefer pytest for skip
- Remove unreachable code
- Copy xxmodule.c from Python 3.11 and 3.8, restoring tests for build_ext.
- Exclude Python 3.11 on macOS due to lack of wheels. Ref pypa/distutils#165.
- Mark test as xfail for now. Ref pypa/distutils#166.
- Use pathlib to read the text
- ⚫ Fade to black.
- Convert TestInstall to pytest
- Only xfail on Windows
- Allow overriding toxworkdir with an env var.
- Ensure sys.version is restored in test_cygwinccompiler. Fixes #166.
- Run test_xx in process, utilizing import_helper On Windows, move the extension module to another temporary directory.
- Redirect extension module to a directory that's not deleted on Windows.
- Include cygwin
- Prefer pytest in test_build_py
- Prefer pytest in test_check
- Refactor imports around docutils.
- Replace addCleanup with monkeypatch.
- Enable tests requiring docutils.
- Fix broken tests around docutils.
- Convert PendingDeprecationWarnings to DeprecationWarnings.
- Ignore unactionable warnings in docutils.
- Prefer pytest for skip
- Prefer pytest for skip
- Prefer pytest for skip
- Prefer pytest for skip
- Prefer pytest for skip
- Convert TestLog to pytest.
- Prefer pytest for skip
- Consolidate tests
- Prefer pytest for skip
- Prefer pytest for skip
- Convert unconditional skip to conditional skip. Mark test as xfail because it's failing.
- Prefer pytest for skip
- ⚫ Fade to black.
- Convert unix_compat to pytest skips
- Consistently import unittest.mock.
- Prefer pytest for skip
- Remove unnecessary comment.
- ⚫ Fade to black.
- 👹 Feed the hobgoblins (delint).
- Prefer tabs
- Add pytest-flake8 and pytest-black and pytest-cov to test lint and style and coverage
- 👹 Feed the hobgoblins (delint).
- Add test capturing failed expectation.
- Add compatibility for Python 3.7
- Windows is sensitive even on Python 3.10
- Also test library dirs
- Extract fixture for c_file
- Generate a C file that imports Python.h and something platform specific.
- Ensure Python include directory is configured.
- Extend the test to compile a second time after setting include dirs again.
- Ensure Windows SDK directories are not cleared when caller specifies include/library dirs
- Remove stray colon
- Fixup bad super() call
- Use CCompiler._fix_compile_args to fix args to compile()
- ⚫ Fade to black.
- Allow compiler classes to supply include and library dirs at the class level.
- Disallow repeat calls to .initialize in one place.




## Summary of changes



Closes 

### Pull Request Checklist
- [x] Changes have tests
- [x] News fragment added in [`changelog.d/`].
  _(See [documentation][PR docs] for details)_


[`changelog.d/`]: https://github.com/pypa/setuptools/tree/master/changelog.d
[PR docs]:
https://setuptools.pypa.io/en/latest/development/developer-guide.html#making-a-pull-request
